tasks/vm.py

```diff
@@ -51,6 +52,9 @@ def create_vm(
     if ami_id is not None:
         extra_flags["ddinfra:osAmiId"] = ami_id
 
+    if private_key_path is not None:
```

Comment:
Can you add instead the ssh key to the ssh agent so the code is the same in both cases?